src/towncrier/test/test_build.py

```diff
@@ -143,6 +143,32 @@ def test_in_different_dir_config_option(self, runner):
         self.assertEqual(0, result.exit_code)
         self.assertTrue((project_dir / "NEWS.rst").exists())
 
+    @with_isolated_runner
+    def test_in_different_dir_with_nondefault_newsfragments_directory(self, runner):
+        """
+        Config location differs from the base directory for news file and fragments.
+
+        This is useful when multiple projects share one towncrier configuration.
+        The default `newsfragments` setting already supports this scenario so here
+        we test that custom settings also do.
+        """
+        Path("pyproject.toml").write_text(
+            "[tool.towncrier]\n" + 'directory = "changelog.d"\n'
+        )
+        Path("foo/foo").mkdir(parents=True)
+        Path("foo/foo/__init__.py").write_text("")
+        Path("foo/changelog.d").mkdir()
+        Path("foo/changelog.d/123.feature").write_text("Adds levitation")
+        self.assertFalse(Path("foo/NEWS.rst").exists())
+
+        result = runner.invoke(
+            cli,
+            ("--yes", "--config", "pyproject.toml", "--dir", "foo", "--version", "1.0"),
```

Comment:
I see normpath was previously used.

Maybe for the test, we can go wild, and add something like this, to make sure `dir` still works when it is passed as a non-normalized path.

```suggestion
            ("--yes", "--config", "pyproject.toml", "--dir", "./foo/../foo/", "--version", "1.0"),
```

also, do we need to pass `--config=pyproject.toml` here ? I was thinking that this is already the default value.

Reply:
Thanks for "wild" suggestion, I'll try it out!

We must explicitly pass `--config` here because otherwise we will search for `pyproject.toml` inside the directory provided by `--dir` :)